### DISCUSSION TITLE
RawDecoderSettings

### DISCUSSION THREAD

**Discussion by davidekholm** [lanthale/LibRawFX#5] [Ideas]
Hi. I'm happy to see that you're working on implementing RawDecoderSettings. That's probably the only thing preventing us from ditching our old libraw integration (using BridJ) and using your library within jAlbum instead. We plan to write a Java ImageIO frontend to your library instead (unless you wish to write it yourself ;-)). Any time estimates on when RawDecoderSettings will be implemented?

**Comment by lanthale** [comment]
I have now implemented RawDecoderSettings. If you have any question, just ask. What is missing is that I translate all the int values the libraw engine is using into better readable values. That means you must know the range of parameter data and how to use it.

**Comment by davidekholm** [comment]
Thanks! I'll hopefully be able to spend time with this during next week. I've attached our current ImageIO frontend to LibRaw in case you like to see what parts of LibRaw we're using. I believe RawDecoderSettings was the missing part. See RAWImageReadParam and LibRawImageReader.read()
[jlibraw-src.zip](https://github.com/lanthale/LibRawFX/files/9731608/jlibraw-src.zip)

**Comment by lanthale** [comment]
Hi,

I have started today and will be finishing end of this week if all goes well. 

Clemens

**Reply by davidekholm** [reply]
Awsome! I stay tuned.

**Reply by davidekholm** [reply]
Hi. Just checking the status? Are you finished with the changes?

**Comment by lanthale** [comment]
Yes I am finished. I just will add an additional interface where you can add a hashmap with cameramodel/setting object. On 14. Nov 2022, at 17:27, davidekholm ***@***.***> wrote:﻿
Hi. Just checking the status? Are you finished with the changes?

**Reply by davidekholm** [reply]
Hi. Ready now? Please let me know as I'm eager to integrate this library into jAlbum. (I take it we'll have to wait for another Java 19 release until this library runs on ARM64 Macs?)

**Comment by davidekholm** [comment]
Great! Just let me know when I should commence integrating this into jAlbum (writing our own ImageIO front-end to it I assume)
